packages/update-monitor/src/common/accessControl.ts

```diff
@@ -0,0 +1,66 @@
+import { providers } from 'ethers'
+
+import { AccessControl__factory } from '../typechain/factories/common/AccessControl__factory'
+import { bytes32ToAddress } from './address'
+
+// many deployments are using OpenZeppelin access control contracts
+// this file includes methods for obtaining the data from them
+// source code of the contract
+// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.3/contracts/access/IAccessControl.sol
+export async function getRoleAdmin(
+  provider: providers.JsonRpcProvider,
+  address: string,
+  role: string,
+) {
+  const accessControl = AccessControl__factory.connect(address, provider)
+
+  return await accessControl.getRoleAdmin(role)
+}
+
+export async function getRoleMembers(
+  provider: providers.JsonRpcProvider,
+  address: string,
+  fromBlock: number,
+  role: string,
+): Promise<string[]> {
+  const accessControl = AccessControl__factory.connect(address, provider)
+
+  const grantedFilter = accessControl.filters.RoleGranted(role)
+
+  const logsGranted = await provider.getLogs({
+    ...grantedFilter,
+    fromBlock,
+  })
+
+  const granted = logsGranted
+    .filter((log) => log.topics[1] === role)
+    .map((log) => log.topics[2])
+
+  const revokedFilter = accessControl.filters.RoleRevoked(role)
+
+  const logsRevoked = await provider.getLogs({
+    ...revokedFilter,
+    fromBlock,
+  })
+
+  const revoked = logsRevoked
+    .filter((log) => log.topics[1] === role)
+    .map((log) => log.topics[2])
```

Comment:
I think it would look simpler to inline logic of removing grantees here and remove `getMembers()` function below completely.

```suggestion
    .map((log) => log.topics[2])
    .forEach((revokedAddress) => {
        const index = granted.indexOf(revokedAddress)
        if (index !== -1) {
            granted.splice(index, 1)
        }
    })
```

But if you prefer to keep `getMembers()` instead then I'm ok with it, but then I'd prefer for it to not modify passed parameter (`revoked`), as it's not a good practice. Personally I prefer the inlined version above.

Reply:
good idea, thanks, I will leave the inline function for now, maybe in the future it would be a good idea to make a pure function out of it for easier testing